Helper/TbkResponseHelper.php

```diff
@@ -0,0 +1,197 @@
+        }
+
+        if ($transactionResult->paymentTypeCode == 'VD') {
+            $paymentType = 'Débito';
+        } elseif ($transactionResult->paymentTypeCode == 'VP') {
+            $paymentType = 'Prepago';
+        } else {
+            $paymentType = 'Crédito';
+        }
+        $installmentsString = '';
+        if ($tipoCuotas != 'Sin cuotas') {
+            $installmentsString = "
+                <div>
+                    • N&uacute;mero de cuotas: <b>{$transactionResult->installmentsNumber}</b>
+                </div>
+                <div>
+                    • Monto Cuota: <b>{$transactionResult->installmentsAmount}</b>
+                </div>
+            ";
+        }
+
+        return "
+            <b>Detalles del pago con {$product}</b>
+            <div>
+                • Respuesta de la Transacci&oacute;n: <b>{$transactionResponse}</b>
+            </div>
+            <div>
+                • C&oacute;digo de la Transacci&oacute;n: <b>{$transactionResult->responseCode}</b>
+            </div>
+            <div>
+                • Monto: <b>$ {$transactionResult->amount}</b>
+            </div>
+            <div>
+                • Order de Compra: <b>$ {$transactionResult->buyOrder}</b>
+            </div>
+            <div>
+                • Fecha de la Transacci&oacute;n: <b>".substr(self::utcToLocalDate($transactionResult->transactionDate), 0,10).'</b>
+            </div>
+            <div>
+                • Hora de la Transacci&oacute;n: <b>'.substr(self::utcToLocalDate($transactionResult->transactionDate),11, 8)."</b>
+            </div>
+            <div>
+                • Tarjeta: <b>**** **** **** {$transactionResult->cardNumber}</b>
+            </div>
+            <div>
+                • C&oacute;digo de autorizacion: <b>{$transactionResult->authorizationCode}</b>
+            </div>
+            <div>
+                • Tipo de Pago: <b>{$paymentType}</b>
+            </div>
+            <div>
+                • Tipo de Cuotas: <b>{$tipoCuotas}</b>
+            </div>
+            {$installmentsString}
+            ";
+    }
+
+    public static function getRejectMessage($transactionResult, $product)
+    {
+        if (strpos($product, 'click')) {
+            $transactionResult = self::getOneclickDetails($transactionResult);
+            return 'Transacción rechazada con Oneclick Mall' .
+                nl2br('• Respuesta de la Transacción: ' . $transactionResult->responseCode . ' ') .
+                nl2br('• Monto:$ ' . $transactionResult->amount . ' ') .
+                nl2br('• Order de Compra: ' . $transactionResult->buyOrder . ' ') .
+                nl2br('• Fecha de la Transacción: ' . substr(self::utcToLocalDate($transactionResult->transactionDate), 0,10) . ' ') .
+                nl2br('• Hora de la Transacción: ' . substr(self::utcToLocalDate($transactionResult->transactionDate),11, 8). ' ') .
+                nl2br('• Tarjeta: **** **** **** ' . $transactionResult->cardNumber . '');
```

Comment:
this isn't the same format used in other messages.

Reply:
is different because Magento does not render the message the same when the transaction is rejected and is not redirected away from the site